synapse/util/caches/cached_call.py

```diff
@@ -65,7 +69,7 @@ def __init__(self, f: Callable[[], Awaitable[TV]]):
         """
         self._callable: Optional[Callable[[], Awaitable[TV]]] = f
         self._deferred: Optional[Deferred] = None
-        self._result: Union[None, Failure, TV] = None
+        self._result: Union[_Sentinel, TV, Failure] = _Sentinel.sentinel
```

Comment:
Looks like `_result` is an `Union[_Sentinel, TV, Failure]` but its value is an `object`? Or am I misunderstanding something here?

Reply:
I think you're missing something. Or I don't understand the question.

`_result` starts out as `_Sentinel.sentinel`, which (via the magic of Enums) is an instance of `_Sentinel`. It is later set to either a `TV` or a `Failure`.